From 733343547b0f3e7303844615749510da9258736a Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Wed, 4 May 2016 16:03:39 +0300
Subject: [PATCH 0329/1345] mvpp2x: tai and ptp structures

commit  560355efa0da50dbebeee5fe7f309b9acc146b1e from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

TAI hw unit has own hw-mapping (in DTS),
so add it to the (struct cpn110_gop_hw)

Add (struct mv_pp2x_ptp_desc *) per-port private data
to (struct mv_pp2x_port)

Signed-off-by: Yan Markman <ymarkman@marvell.com>
Change-Id: Id9fd38598870574ebdb8e3a723be8e5a719c820c
Reviewed-on: http://vgitil04.il.marvell.com:8080/29632
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Yelena Krivosheev <yelena@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
index c8698b0..b24bc73 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
@@ -390,6 +390,7 @@ struct cpn110_gop_hw {
 	struct mv_mac_unit_desc xlg_mac;
 	struct mv_mac_unit_desc serdes;
 	struct mv_mac_unit_desc xmib;
+	struct mv_mac_unit_desc tai;
 	struct mv_mac_unit_desc ptp;
 	void __iomem *smi_base;
 	void __iomem *xsmi_base;
@@ -538,6 +539,8 @@ struct queue_vector {
 	struct mv_pp2x_port *parent;
 };
 
+struct mv_pp2x_ptp_desc; /* per-port private PTP descriptor */
+
 struct mv_pp2x_port {
 	u8 id;
 
@@ -589,6 +592,8 @@ struct mv_pp2x_port {
 	 * mv_pp2_isr(int irq, void *dev_id=q_vector)
 	 */
 	struct queue_vector q_vector[MVPP2_MAX_CPUS + MVPP2_MAX_SHARED];
+
+	struct mv_pp2x_ptp_desc *ptp_desc;
 };
 
 struct pp2x_hw_params {
-- 
1.7.9.5

